Container Security/kernel-eks-implementation-guide.md

```diff
@@ -0,0 +1,242 @@
+   Using default tag: latest
+   The push refers to repository [123456789123.dkr.ecr.eu-west-1.amazonaws.com/falcon-node-sensor]
+   913513e672ba: Pushed
+   2c5b2be1a19c: Pushed
+   88cd0d79914e: Pushed
+   c20d459170d8: Pushed
+   db978cae6a05: Pushed
+   aeb3f02e9374: Pushed
+   latest: digest: sha256:132eea2728db09c49fecfae78778f11225f6e9818c71c4f5b2321a0dae4d0c95 size: 1572
+   ```
+
+### Step 5: Deploy the DaemonSet using the helm chart
+
+ - Provide CrowdStrike Falcon Customer ID as environment variable. This CID will be used be helm chart to register your cluster nodes to the CrowdStrike Falcon platform.
+   ```
+   $ CID=1234567890ABCDEFG1234567890ABCDEF-12
+   ```
+
+ - Add the CrowdStrike Falcon Helm repository
+   ```
+   $ helm repo add crowdstrike https://crowdstrike.github.io/falcon-helm
+   ```
+
+ - Install CrowdStrike Falcon Helm Chart. Above command will install the CrowdStrike Falcon Helm Chart with the release name falcon-helm in the falcon-system namespace.
+   ```
+   $ helm upgrade --install falcon-helm crowdstrike/falcon-sensor \
+        -n falcon-system --create-namespace \
+        --set falcon.cid="$CID" \
+        --set node.image.repository=$FALCON_NODE_IMAGE_URI
+   Release "falcon-helm" does not exist. Installing it now.
+   NAME: falcon-helm
+   LAST DEPLOYED: Fri Mar  5 17:07:54 2021
+   NAMESPACE: falcon-system
+   STATUS: deployed
+   REVISION: 1
+   TEST SUITE: None
+   NOTES:
+   You should be a Crowdstrike customer and have access to the Falcon Linux Sensor
+   and Falcon Container Downloads to install this helm chart and have it work
+   correctly as a specialized container has to exist in the container registry
+   before this chart will install properly.
+
+   The CrowdStrike Falcon sensor should be spinning up on all your Kubernetes nodes
+   now. There should be no further action on your part unless no Falcon Sensor
+   container exists in your registry. If you forgot to add a Falcon Sensor image to
+   your image registry before you ran `helm install`, please add the Falcon Sensor
+   now; otherwise, pods will fail with errors and crash until there is a valid
+   image to pull. The default image name to deploy a kernel sensor to a node is
+   `falcon-node-sensor`.
+   ```
+   To learn more about falcon-helm visit [upstream github](https://github.com/CrowdStrike/falcon-helm).
+
+ - (optional) Verify that falcon-node-sensor pod running. You should see one pod per each node on your cluster.
+   ```
+   $ kubectl get pods -n falcon-system
+   NAME                              READY   STATUS    RESTARTS   AGE
+   falcon-helm-falcon-sensor-bs98m   2/2     Running   0          21s
+   ```
+ - (optional) Verify that given pod has registered with CrowdStrike Falcon and received unique identifier.
+   ```
+   $ kubectl exec -n falcon-system falcon-helm-falcon-sensor-bs98m -c falcon-node-sensor -- falconctl -g --aid
+   aid="a582XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+   ```
```

Comment:
could/should we write a one liner that iterates through the `kubectl get pods` command, passes them to this `exec` command, and returns all the AIDs?

Reply:
Otherwise the use will copy/paste all the kube cluster names. Could get to be a lot of copy/pasting.